packages/web-components/src/components/va-file-input/va-file-input.tsx

```diff
@@ -122,6 +158,54 @@ export class VaFileInput {
     return this.buttonText ? this.buttonText : 'Upload file';
   };
 
+  /**
+   * Converts the size of a file from bytes to a more human-readable format for
+   * rendering the file size label. This function calculates the file size in
+   * appropriate units (B, KB, MB, GB, TB) based on the size provided. It uses
+   * logarithmic scaling to determine the unit, then formats the size to one
+   * decimal place for units KB and above.
+   *
+   * @param {number} filesSize - The size of the file in bytes.
+   * @returns {string} The file size formatted as a string with the appropriate unit.
+   */
+  private formatFileSize = (filesSize): string => {
+    const units = ['B', 'KB', 'MB', 'GB', 'TB'];
```

Comment:
I really hope no one tries to upload a file that is x GBs or TBs large lol

Reply:
Indeed, that would be quite an edge case! However, it might be wise to consider implementing a limit on file size or handling such scenarios gracefully to prevent potential issues. What are your thoughts?